crates/ide_assists/src/handlers/extract_function.rs

```diff
@@ -634,7 +627,22 @@ fn extraction_target(node: &SyntaxNode, selection_range: TextRange) -> Option<Fu
 
 /// list local variables that are referenced in `body`
 fn vars_used_in_body(ctx: &AssistContext, body: &FunctionBody) -> Vec<Local> {
-    // FIXME: currently usages inside macros are not found
+    let vars_in_macro_call = body.descendants()
+        .filter_map(ast::MacroCall::cast)
+        .filter_map(|ast| ctx.sema.expand(&ast))
+        .flat_map(|ast| ast.descendants())
+        .filter_map(ast::NameRef::cast)
+        .filter_map(|name_ref| NameRefClass::classify(&ctx.sema, &name_ref))
+        .map(|name_kind| match name_kind {
+            NameRefClass::Definition(def) => def,
+            NameRefClass::FieldShorthand { local_ref, field_ref: _ } => {
+                Definition::Local(local_ref)
+            }
+        })
+        .filter_map(|definition| match definition {
+            Definition::Local(local) => Some(local),
+            _ => None,
+        });
```

Comment:
Does `sema.expand` expand all underlying macros for an invocation or only one level? I think it only goes one level deep right? In which case this doesn't fully fix the problem.